pycodestyle.py

```diff
@@ -1637,41 +1660,27 @@ def python_3000_invalid_escape_sequence(logical_line, tokens, noqa):
     if noqa:
         return
 
-    # https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals
-    valid = [
-        '\n',
-        '\\',
-        '\'',
-        '"',
-        'a',
-        'b',
-        'f',
-        'n',
-        'r',
-        't',
-        'v',
-        '0', '1', '2', '3', '4', '5', '6', '7',
-        'x',
-
-        # Escape sequences only recognized in string literals
-        'N',
-        'u',
-        'U',
-    ]
-
     prefixes = []
     for token_type, text, start, _, _ in tokens:
-        if token_type in {tokenize.STRING, FSTRING_START, TSTRING_START}:
+        if (
+                token_type == tokenize.STRING or
+                token_type == FSTRING_START or
+                token_type == TSTRING_START
+        ):
```

Comment:
I am watching your stream VOD and I simply wanted to let you know that the reason the previous solution was very slow is that Python LOAD_CONST optimization applies only to trivial Python objects (with guaranteed no side effects). `tokenize.STRING` may have side effects, which prevents this optimization and makes Python construct the set each time you access it in a conditional. My typical workaround for such performance-oriented cases, without polluting the global scope with lots of sets/dicts, is to simply put them in kwargs of the function that run just once. See this bytecode: https://godbolt.org/z/qoPdYvha3